source/server/proto_descriptors.cc

```diff
@@ -42,8 +41,10 @@ TEST(ProtoDescriptorTest, BackCompat) {
   };
 
   for (const auto& method : methods) {
-    EXPECT_NE(Protobuf::DescriptorPool::generated_pool()->FindMethodByName(method), nullptr)
-        << fmt::format("{} not found", method);
+    if (Protobuf::DescriptorPool::generated_pool()->FindMethodByName(method) == nullptr) {
+      descriptors_valid = false;
```

Comment:
nit: just return false immediately. You can delete the local variable, and then just return true at the end of the function.